### PR TITLE
Fix watcher subscription sync and startup config loading

### PR DESCRIPTION
**Pull request by agatx**
## Summary
- load `ALLOWED_ORIGINS` after `.env` initialization and add regression coverage for configured, same-host, and localhost CORS behavior
- make `watch_rooms` authoritative on the server, align web and iOS room watchers with clear-on-empty and host-change behavior, and add watcher regression tests
- prune idle public rate-limit buckets and update protocol/package/PRD docs to reflect current multiparty and diagnostics behavior

## Testing
- `cd server && go test ./...`
- `cd client && npm test`
- `cd client-android && ./gradlew :serenada-core:testDebugUnitTest :app:testDebugUnitTest`
- `cd client-ios && xcodegen generate && xcodebuild -project SerenadaiOS.xcodeproj -scheme SerenadaiOS -destination 'platform=iOS Simulator,name=iPhone 16,OS=18.4' test`
  - `SerenadaiOSTests` unit suite passed, including the new `RoomWatcherTests`
  - existing UI tests timed out waiting for participant count to rise above 1 in `DeepLinkParticipantCountUITests` and `DeepLinkRejoinFlowUITests`

### CODE REVIEW

File: client/packages/core/src/RoomWatcher.ts [hunk 1]
```diff
@@ -68,6 +68,9 @@ export class RoomWatcher {
         this.notify();
 
         if (this.watchedRoomIds.length === 0) {
+            if (this.hasConnected && this.signaling.isConnected) {
+                this.signaling.watchRooms([]);
+            }
             return;
         }
```
[the remark this answers is not in the document]
Comment (agatx) [reply client/packages/core/src/RoomWatcher.ts:76]: Fixed in 2b60a73. `SignalingEngine.watchRooms` now sends `watch_rooms` even when `rids` is empty, and the web signaling tests now cover that transport-level behavior directly.

File: server/multiparty_test.go [hunk 2]
```diff
@@ -518,22 +531,120 @@ func TestWatchRoomsIncludesMaxParticipants(t *testing.T) {
...
+	hub.handleMessage(watcher, watchRoomsPayload([]string{}))
+	msgs := drainMessages(watcher)
+	for _, msg := range msgs {
+		if msg.Type != "room_statuses" {
+			continue
+		}
+		var payload map[string]map[string]int
+		if err := json.Unmarshal(msg.Payload, &payload); err != nil {
+			t.Fatalf("failed to parse room_statuses payload: %v", err)
+		}
```
[the remark this answers is not in the document]
Comment (agatx) [reply server/multiparty_test.go:630]: Fixed in 2b60a73. This test now requires an explicit `room_statuses` response when clearing the watch list, so it can no longer pass silently if that server response regresses.

File: client/packages/core/src/RoomWatcher.ts [hunk 3]
```diff
@@ -68,6 +68,9 @@ export class RoomWatcher {
         this.notify();
 
         if (this.watchedRoomIds.length === 0) {
+            if (this.hasConnected && this.signaling.isConnected) {
+                this.signaling.watchRooms([]);
+            }
```
[the remark this answers is not in the document]
Comment (agatx) [reply client/packages/core/src/RoomWatcher.ts:73]: Fixed in 2b60a73. `SignalingEngine.watchRooms` no longer drops empty arrays, so `watch_rooms` with `rids: []` now reaches the transport. I also added a direct `SignalingEngine` regression test for the clear-on-empty path.

File: server/multiparty_test.go [hunk 4]
```diff
@@ -518,22 +531,120 @@ func TestWatchRoomsIncludesMaxParticipants(t *testing.T) {
...
+	drainMessages(watcher)
+
+	hub.handleMessage(watcher, watchRoomsPayload([]string{ridB}))
+	msgs := drainMessages(watcher)
+	for _, msg := range msgs {
+		if msg.Type != "room_statuses" {
+			continue
+		}
+		var payload map[string]map[string]int
+		if err := json.Unmarshal(msg.Payload, &payload); err != nil {
```
[the remark this answers is not in the document]
Comment (agatx) [reply server/multiparty_test.go:554]: Fixed in 2b60a73. This test now tracks whether a `room_statuses` message was observed and fails if the server stops emitting the bulk response when replacing watched rooms.
